src/01-dinosaur-facts.js

```diff
@@ -71,7 +89,20 @@ function getDinosaurDescription(dinosaurs, id) {}
  *  getDinosaursAliveMya(dinosaurs, 65, "unknown-key");
  *  //> ["WHQcpcOj0G"]
  */
-function getDinosaursAliveMya(dinosaurs, mya, key) {}
+ function getDinosaursAliveMya(dinosaurs, mya, key) {
+  let dinosaursArray = [] 
+  for (let i = 0; i < dinosaurs.length; i++) {
+    if ((mya <= dinosaurs[i].mya[0] && mya >= dinosaurs[i].mya[1]) || ((mya  === dinosaurs[i].mya[0])) || (dinosaurs[i].mya - mya === 1)) { 
```

Comment:
Tough logic to handle. Well done! You clearly have thought a lot about how to combine conditions.

Same note as above regarding brackets. Especially with a nested `if`, like with this one!